src/main/java/com/flagsmith/FlagsmithCachedAPIWrapper.java

```diff
@@ -13,12 +16,16 @@ public FlagsmithCachedAPIWrapper(final FlagsmithCacheConfig.FlagsmithInternalCac
 
   @Override
   public FlagsAndTraits getFeatureFlags(FeatureUser user, boolean doThrow) {
-    if (user == null) {
-      // not caching project flags yet
+    String cacheKey = cache.getProjectFlagsCacheKey();
```

Comment:
I think I prefer something like: 

```java
String projectFlagsCacheKey = cache.getProjectFlagsCacheKey();
if (...) {
  ...
}
if (user != null) {
  assertValidUser(user);
  return cache.getCache().get(user.getIdentifier(), k -> ...)
}
return cache.getCache().get(projectFlagsCacheKey, k -> ...)
```

Maybe it's my python leaning tendencies but I mainly don't like the reuse of the `cacheKey` variable for 2 separate purposes, even though they kind of achieve the same thing. 

Happy to discuss this further though, definitely not totally against either approach.

Reply:
Do you mean to replicate line `return cache.getCache().get(projectFlagsCacheKey, k -> ...)`? I don't like duplicating code and that line is quite important. Would you prefer creating a new private method to do this?

Reply:
It's not replicating the line though? It's a _similar_ line, but with a different input. I agree that we shouldn't replicate code, but I also don't think we should overwrite an existing variable with a totally different value.